src/test/java/seedu/rc4hdb/storage/JsonAdaptedResidentTest.java

```diff
@@ -0,0 +1,184 @@
+package seedu.rc4hdb.storage;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static seedu.rc4hdb.storage.JsonAdaptedResident.MISSING_FIELD_MESSAGE_FORMAT;
+import static seedu.rc4hdb.testutil.Assert.assertThrows;
+import static seedu.rc4hdb.testutil.TypicalResidents.BENSON;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.rc4hdb.commons.exceptions.IllegalValueException;
+import seedu.rc4hdb.model.resident.fields.Email;
+import seedu.rc4hdb.model.resident.fields.Gender;
+import seedu.rc4hdb.model.resident.fields.House;
+import seedu.rc4hdb.model.resident.fields.MatricNumber;
+import seedu.rc4hdb.model.resident.fields.Name;
+import seedu.rc4hdb.model.resident.fields.Phone;
+import seedu.rc4hdb.model.resident.fields.Room;
+
+public class JsonAdaptedResidentTest {
+    private static final String INVALID_NAME = "R@chel";
+    private static final String INVALID_PHONE = "+651234";
+    private static final String INVALID_EMAIL = "example.com";
+    private static final String INVALID_ROOM = "03_08";
+    private static final String INVALID_GENDER = "m";
+    private static final String INVALID_HOUSE = "d";
+    private static final String INVALID_MATRIC_NUMBER = "a0123456a";
+    private static final String INVALID_TAG = "#friend";
+
+    private static final String VALID_NAME = BENSON.getName().toString();
+    private static final String VALID_PHONE = BENSON.getPhone().toString();
+    private static final String VALID_EMAIL = BENSON.getEmail().toString();
+    private static final String VALID_ROOM = BENSON.getRoom().toString();
+    private static final String VALID_GENDER = BENSON.getGender().toString();
+    private static final String VALID_HOUSE = BENSON.getHouse().toString();
+    private static final String VALID_MATRIC_NUMBER = BENSON.getMatricNumber().toString();
+    private static final List<JsonAdaptedTag> VALID_TAGS = BENSON.getTags().stream()
+            .map(JsonAdaptedTag::new)
+            .collect(Collectors.toList());
+
+    @Test
```

Comment:
I assume these tests are taken from the person implementation tests. But nevertheless, the tests are excellent and sufficient. Looking good Jordan.